src/ansi-c/c_typecheck_code.cpp

```diff
@@ -847,7 +847,23 @@ void c_typecheck_baset::typecheck_dowhile(code_dowhilet &code)
   }
 }
 
-void c_typecheck_baset::typecheck_spec_assigns_condition(exprt &condition)
+void c_typecheck_baset::throw_on_side_effects(const exprt &expr)
```

Comment:
Do we have anything in regression testing for side effects in expressions in the context of assigns clauses?

Reply:
Yes side effects in assigns clause targets are rejected and we have tests for that